### ISSUE TITLE
403 error occurs when Form-Authentication succeeded PAYARA-1244

### ISSUE THREAD

**Issue by MasatoshiTada** [payara/Payara#1213]
# Description #
----------

I created simple Form-Authentication sample (below).
https://github.com/MasatoshiTada/form-authentication-sample
When username and password are valid, 403 error occurs.

## Expected Outcome

When username and password are valid, my sample will show `protected.jsp`.
(About valid username and password, please look .sql file below.)
![2016-11-30 09 48 09](https://cloud.githubusercontent.com/assets/7044137/20735326/bbf0a9f0-b6e2-11e6-8f02-5c720feae10c.png)
![2016-11-30 09 48 21](https://cloud.githubusercontent.com/assets/7044137/20735369/f682e376-b6e2-11e6-8202-b170fa9967cc.png)

Screenshots above are case of Payara 162.
Payara 164 is configured properly (JDBC resource, realm, and so on), because my sample works in Payara 162 (like above) and I use same setup-script (below) to configure Payara 162 and Payara 164.

## Current Outcome

When username and password are valid, application shows `/WEB-INF/403.jsp`.
![2016-11-30 09 49 10](https://cloud.githubusercontent.com/assets/7044137/20735354/e2674ecc-b6e2-11e6-9e8f-77d1cf3fb941.png)

## Steps to reproduce (Only for bug reports) 

1 -**  Install Payara from ZIP file
```bash
unzip payara-web-ml-4.1.1.164.zip
mv payara41 payara164
```

2 -** Run setup script
Please use .sh file below. This file is PROJECT_ROOT/setup-payara.sh.
I use this .sh file to configure Payara 162, too.

```bash
# PAYARA_HOME=/path/to/payara (162 or 164)
export PAYARA_HOME=~/Java/ap-server/payara164

# copy JDBC driver
cp ~/Java/JDBC/postgresql-9.4.1212.jar $PAYARA_HOME/glassfish/lib

$PAYARA_HOME/bin/asadmin start-domain domain1

# create JDBC pool
$PAYARA_HOME/bin/asadmin create-jdbc-connection-pool --datasourceclassname org.postgresql.xa.PGXADataSource --restype javax.sql.XADataSource --property url=jdbc\\:postgresql\\://localhost\\:5432/ex:user=user:password=password ExPool

# ping
$PAYARA_HOME/bin/asadmin ping-connection-pool ExPool

# create JDBC resource
$PAYARA_HOME/bin/asadmin create-jdbc-resource --connectionpoolid ExPool jdbc/ex

# enable default-principal-to-role-mapping
$PAYARA_HOME/bin/asadmin set server-config.security-service.activate-default-principal-to-role-mapping=true

# create realm
$PAYARA_HOME/bin/asadmin create-auth-realm --classname com.sun.enterprise.security.auth.realm.jdbc.JDBCRealm \
--property jaas-context=jdbcRealm:datasource-jndi=jdbc/ex:user-table=users:\
user-name-column=login_id:password-column=encrypt_password:group-table=user_roles:\
group-name-column=role_name:digest-algorithm=SHA-256:\
digestrealm-password-enc-algorithm=digest-algorithm:encoding=Hex:charset=UTF-8 customer

$PAYARA_HOME/bin/asadmin stop-domain domain1
```

3 -** Prepare PostgreSQL

I use PostgreSQL 9.4.4.
Please use .sql file below.

```sql
CREATE TABLE users
(
    login_id VARCHAR(128) PRIMARY KEY,
    encrypt_password VARCHAR(128),
    name VARCHAR(128)
);

CREATE TABLE user_roles
(
    login_id VARCHAR(128),
    role_name VARCHAR(128)
);

-- password is "password"
INSERT INTO users VALUES('adminA', '5e884898da28047151d0e56f8dc6292773603d0d6aabbdd62a11ef721d1542d8', 'adminA');
INSERT INTO users VALUES('adminB', '5e884898da28047151d0e56f8dc6292773603d0d6aabbdd62a11ef721d1542d8', 'adminB');

INSERT INTO user_roles(login_id, role_name) VALUES('adminA', 'ADMINISTRATOR');
INSERT INTO user_roles(login_id, role_name) VALUES('adminB', 'ADMINISTRATOR');
```

4 -** Build, deploy, run

```bash
mvn clean package
# deploy .war with asadmin or IDE
```

5 -** Access

Access https://localhost:8181/sample/ with the browser, and input **valid** username and password.
USERNAME is "adminA" or "adminB", and PASSWORD is "password".
![2016-11-30 09 48 09](https://cloud.githubusercontent.com/assets/7044137/20736009/26af19d0-b6e7-11e6-877b-63bf6df35292.png)

6 -** Result

Browser shows `/WEB-INF/403.jsp`(configured in `web.xml` `<error-page>` of 403 error).
![2016-11-30 10 11 50](https://cloud.githubusercontent.com/assets/7044137/20735753/8ca05b0c-b6e5-11e6-9931-d00ac96f762b.png)

7 -** Case of **invalid** username and password

Browser shows `/WEB-INF/login_error.jsp`(configured in `web.xml` `<form-error-page>`).
This is correct behavior. Of course, it works in Payara 162, too.
![2016-11-30 10 19 45](https://cloud.githubusercontent.com/assets/7044137/20735901/89eb1f4a-b6e6-11e6-897a-85c07d89de97.png)

### Samples

https://github.com/MasatoshiTada/form-authentication-sample

.sql file and .sh file are included in the project root.

## Environment ##

- **Payara Version**: 4.1.1.164 (This bug does not occur in Payara 162)
- **Edition**: Web ML
- **JDK Version**: Oracle JDK8u112
- **Operating System**: macOS Sierra, Windows 7

**Comment by smillidge** [comment]
This is a known issue the default role mapping security setting is broken in 164. This will be fixed soon.

This has no effect in 164 it is a bug. $PAYARA_HOME/bin/asadmin set server-config.security-service.activate-default-principal-to-role-mapping=true

You can add a role mapping into glassfish-web.xml or add the default role mapping setting as discussed here.
https://payara.gitbooks.io/payara-server/content/documentation/extended-documentation/app-deployment/descriptor-elements.html

**Comment by mikecroft** [comment]
`PAYARA-1244`

**Comment by MasatoshiTada** [comment]
Thank you so much!
I added glass fish-web.xml, and I found Form-Authentication succeeded properly.
https://github.com/MasatoshiTada/form-authentication-sample/commit/40ede4f660400713895a1d93f2818fd2f279ca24

**Comment by smillidge** [comment]
Fixed by #1218
